debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/EventBuffer.java

```diff
@@ -0,0 +1,141 @@
+package io.debezium.connector.mysql;
+
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.shyiko.mysql.binlog.event.Event;
+import com.github.shyiko.mysql.binlog.event.EventType;
+import com.github.shyiko.mysql.binlog.event.QueryEventData;
+
+import io.debezium.connector.mysql.BinlogReader.BinlogPosition;
+
+class EventBuffer {
+    private static final Logger logger = LoggerFactory.getLogger(EventBuffer.class);
+
+
+    private final int capacity;
+    private final Queue<Event> buffer;
+    private final BinlogReader reader;
+    private boolean txStarted = false;
+    private BinlogPosition largeTxNotBufferedPosition;
```

Comment:
Some comments on the class and also non-obvious fields like this would be great. It seems that's the position where we stopped buffering?